master/internal/db/postgres_users.go

```diff
@@ -425,3 +427,32 @@ func (db *PgDB) AuthTokenKeypair() (*model.AuthTokenKeypair, error) {
 		return &tokenKeypair, nil
 	}
 }
+
+// UpdateUserSetting updates user setting.
+func UpdateUserSetting(setting *model.UserWebSetting) error {
+	if len(setting.Value) == 0 {
+		_, err := Bun().NewDelete().Model(setting).Where(
+			"user_id = ?", setting.UserID).Where(
+			"storage_path = ?", setting.StoragePath).Where(
+			"key = ?", setting.Key).Exec(context.TODO())
+		return err
+	}
+
+	_, err := Bun().NewInsert().Model(setting).On("CONFLICT (user_id, key, storage_path) DO UPDATE").
+		Set("value = EXCLUDED.value").Exec(context.TODO())
+	return err
+}
+
+// GetUserSetting gets user setting.
+func GetUserSetting(userID model.UserID) ([]*userv1.UserWebSetting, error) {
+	setting := []*userv1.UserWebSetting{}
+	err := Bun().NewSelect().Model(&setting).Where("user_id = ?", userID).Scan(context.TODO())
+	return setting, err
+}
+
+// ResetUserSetting resets user setting.
+func ResetUserSetting(userID model.UserID) error {
+	var setting model.UserWebSetting
+	_, err := Bun().NewDelete().Model(&setting).Where("user_id = ?", userID).Exec(context.TODO())
```

Comment:
question: would it make sense for reset to set the settings to empty as opposed to deleting the row (don't really have a definite idea here, just was wondering)

Reply:
This was trying to align with front end storage, when we `clear` the local storage, we remove everything related, so at the backend, I wanted to do the same.